policybot/pkg/storage/spanner/query.go

```diff
@@ -67,6 +67,51 @@ func (s store) QueryIssuesByRepo(context context.Context, orgID string, repoID s
 	return err
 }
 
+func (s store) QueryTestResultByName(context context.Context, testName string, cb func(*storage.TestResult) error) error {
+	iter := s.client.Single().Query(context, spanner.Statement{SQL: fmt.Sprintf("SELECT * FROM TestResults WHERE TestName = '%s'", testName)})
+	err := iter.Do(func(row *spanner.Row) error {
+		testResult := &storage.TestResult{}
+		if err := row.ToStruct(testResult); err != nil {
+			return err
+		}
+
+		return cb(testResult)
+	})
+
+	return err
+}
+
+func (s store) QueryTestResultByPrNumber(context context.Context, prNum int64, cb func(*storage.TestResult) error) error {
+	iter := s.client.Single().Query(context, spanner.Statement{SQL: fmt.Sprintf("SELECT * FROM TestResults WHERE PrNum = '%v'", prNum)})
+	err := iter.Do(func(row *spanner.Row) error {
+		testResult := &storage.TestResult{}
+		if err := row.ToStruct(testResult); err != nil {
+			return err
+		}
+
+		return cb(testResult)
+	})
+
+	return err
+}
+
+func (s store) QueryTestResultByUndone(context context.Context) ([]*storage.TestResult, error) {
+	iter := s.client.Single().Query(context, spanner.Statement{SQL: fmt.Sprintf("SELECT * FROM TestResults WHERE Done = false")})
+	testResults := []*storage.TestResult{}
+	if err := iter.Do(func(row *spanner.Row) error {
+		testResult := &storage.TestResult{}
+		if err := row.ToStruct(testResult); err != nil {
+			return fmt.Errorf("error in fetching undone test result, %v", err)
```

Comment:
At this layer, just return the Spanner error and let the higher level code produce an extended error message.